File: my-app/src/app/common/match-dto.ts
import {ClubDto} from "./club-dto";

export class MatchDto {
  id: number;
  homeTeam: ClubDto;
  awayTeam: ClubDto;
  matchDate: Date;
  result: string;
  homeTeamScore: number;
  awayTeamScore: number;

  constructor(
    id: number,
    homeTeam: ClubDto,
    awayTeam: ClubDto,
    matchDate: Date,
    result: string,
    homeTeamScore: number,
    awayTeamScore: number
  ) {
    this.id = id;
    this.homeTeam = homeTeam;
    this.awayTeam = awayTeam;
    this.matchDate = matchDate;
    this.result = result;
    this.homeTeamScore = homeTeamScore;
    this.awayTeamScore = awayTeamScore;
  }
}

File: FootballApp/src/main/java/com/football/dev/footballapp/dto/MatchDTO.java
package com.football.dev.footballapp.dto;

import com.football.dev.footballapp.models.Club;
import com.football.dev.footballapp.models.Stadium;
import com.football.dev.footballapp.models.enums.MatchStatus;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class MatchDTO{
    private Long id;
    private ClubDto homeTeam;
    private ClubDto awayTeam;
    private LocalDateTime matchDate;
//    private Stadium stadium;
    private String result;
    private Integer homeTeamScore;
    private Integer awayTeamScore;
    private MatchStatus matchStatus;

}
